src/main/java/org/esbtools/gateway/resync/exception/IncompleteRequestException.java

```diff
@@ -0,0 +1,15 @@
+package org.esbtools.gateway.resync.exception;
+
+import org.esbtools.gateway.resync.ResyncRequest;
+import org.esbtools.gateway.resync.service.ResyncErrorMessages;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value= HttpStatus.BAD_REQUEST, reason=ResyncErrorMessages.INCOMPLETE_REQUEST)
```

Comment:
Hrm... so I think we're mixing Spring exception handling strategies a little bit. I don't think these annotations are necessary if you're using the `@ExceptionHandler` methods. I think the exception handler methods are probably the way to go, judging by this warning in the Spring docs for `@ResponseStatus` exceptions:

> Warning: when using this annotation on an exception class, or when setting the reason attribute of this annotation, the HttpServletResponse.sendError method will be used.

> With HttpServletResponse.sendError, the response is considered complete and should not be written to any further. Furthermore, the Servlet container will typically write an HTML error page therefore making the use of a reason unsuitable for REST APIs. For such cases it is preferable to use a ResponseEntity as a return type and avoid the use of @ResponseStatus altogether.